caddyhttp/proxy/upstream.go

```diff
@@ -320,13 +329,17 @@ func parseBlock(c *caddyfile.Dispenser, u *staticUpstream) error {
 
 func (u *staticUpstream) healthCheck() {
 	for _, host := range u.Hosts {
-		hostURL := host.Name + u.HealthCheck.Path
-		if r, err := u.HealthCheck.Client.Get(hostURL); err == nil {
-			io.Copy(ioutil.Discard, r.Body)
-			r.Body.Close()
-			host.Unhealthy = r.StatusCode < 200 || r.StatusCode >= 400
+		if host.Dynamic {
+			host.Unhealthy = false //dynamic hosts can never be unhealthy
```

Comment:
Okay. I need to see the problem the way you are seeing it. I'll come back to this after I've cleared my head. This might be OK, I just need to convince myself of that.